### ISSUE TITLE
En el formulario de registro no aparece la casilla para aceptar las condiciones

### ISSUE THREAD

**Issue by nosolored** [chamilo/chamilo-lms#2804]
**Describe el problema**
En el formulario de registro de usuarios debería aparecer un campo para aceptar la política de protección de datos, pero no es así.
Cuando el usuario ya se ha registrado y accede al aula, sí le aparece el texto para que lo acepte, pero no debería ser así, ya que la aceptación debe hacerse en el momento de suministrar los datos personales, no después.

**Cómo reproducir**
1. En la versión 1.11.8 de Chamilo hemos marcado la opción de Habilitar Términos y Condiciones en los Parámetros de Configuración de Chamilo. También tenemos habilitado el registro de usuarios.
2. Y luego en el apartado de Términos y condiciones de la administración incluimos un texto de protección de datos y marcamos la opción de HTML. 
3. Salimos de la plataforma y accedemos al enlace de registro. Aparece el formulario pero sin casilla de aceptación de las condiciones
4. Una vez enviado el formulario, accedemos a la plataforma con los datos insertados, y sí nos aparece un texto para aceptar las condiciones

**Comportamiento esperado**
La aceptación de la política de protección de datos debe hacerse en el momento de suministrar los datos, no después

**Contexto adicional**
Lo hemos probado en una plataforma nuestra recién actualizada de la 1.11.8 y también lo hemos hecho aquí: https://11.chamilo.org
El funcionamiento es el mismo con cualquier navegador.

**Comment by jmontoyaa** [comment]
En el paso 3, cuando entro con un nuevo usuario me aparece este formulario:

![selection_001](https://user-images.githubusercontent.com/158935/53720120-3e07ec80-3e60-11e9-90f2-398270ca262c.png)

No entiendo cual es el problema, cuando se usa HTML el texto es:

"By clicking on 'Register' below you are agreeing to the Terms and Conditions".

**Comment by nosolored** [comment]
Esa pantalla aparece cuando ya has accedido con tu usuario y contraseña, es decir, que ya has enviado previamente tus datos al sistema.
La aceptación de las condiciones debe estar en el formulario de registro y no después.
Para los usuarios que ya estuvieran registrados o en el caso de que se modifiquen las condiciones sí es correcto que aparezca esta pantalla, pero si eres un nuevo usuario y estás completando el formulario de registro, la aceptación de las condiciones debería salir en ese mismo formulario, es decir, aquí: https://11.chamilo.org/main/auth/inscription.php

Pero ahí no sale.
Hemos visto que aparece si se activa una opción en el configuration.php ($_configuration['show_conditions_to_user']) pero para que aparezcan hay que insertar las condiciones en el propio configuration.php, y estas condiciones no parecen tener relación con las condiciones que se almacenan en la Administración.

**Comment by jmontoyaa** [comment]
> Esa pantalla aparece cuando ya has accedido con tu usuario y contraseña, es decir, que ya has enviado previamente tus datos al sistema.

Si, exacto.

> La aceptación de las condiciones debe estar en el formulario de registro y no después.
> Para los usuarios que ya estuvieran registrados o en el caso de que se modifiquen las condiciones sí es correcto que aparezca esta pantalla, pero si eres un nuevo usuario y estás completando el formulario de registro, la aceptación de las condiciones debería salir en ese mismo formulario, es decir, aquí: https://11.chamilo.org/main/auth/inscription.php

Bueno actualmente no funciona así. Fue desarrollado para que funcione de esa manera.
No creo que es un bug, pero un feature request.

Lo de "show_conditions_to_user" es exactamente otra opción de configurar los términos y condiciones.

**Comment by nosolored** [comment]
Ok, gracias por la aclaración.
En el Chamilo Talk en que Nicolas hablaba de este plugin indicaba que la aceptación aparecía en el formulario de registro, por eso habíamos pensado que era un bug:
https://www.youtube.com/watch?v=moTi0e8mA1Q

**Comment by NicoDucou** [comment]
Lo que indique en el Chamilo Talk es que hay la posibilidad de aggregar un check box en la pagina de registro para acceptar condiciones con un texto resumido y simple (no el mismo texto que lo que hay en los terminos y condiciones de la pagina de administracion. Justamente estuve pensando en este caso en la opcion 'show_conditions_to_user' que no es lo mismo que lo de la administracion como Julio lo indico.

**Comment by nosolored** [comment]
Muchas gracias Nicolas por la aclaración. 
Para cumplir con el RGPD es necesario que la información sobre protección de datos se suministre en el momento en que se obtienen esos datos, por eso habíamos reportado esta incidencia, pero ya que sabemos su funcionamiento lo aplicaremos utilizando la opción de configuration.php

**Comment by ywarnier** [comment]
Sí, tenis en cuenta  que no podemos modificar mucho la base de datos por ahora. Yo creo que esta tarea debería quedar para la 2.0.

**Comment by christianbeeznest** [comment]
El checkbox para aceptar los términos y condiciones ya está implementado en el formulario de registro. Esto se realizó en el commit https://github.com/chamilo/chamilo-lms/commit/f813d40e5f0e03fd9a8e762475f27549d302d1f6 , donde se añadió el elemento legal_accept con el texto "I have read and agree to the Terms and Conditions" y se marcó como campo obligatorio. Esto garantiza que los usuarios acepten los términos al momento de registrarse, cumpliendo con el comportamiento esperado.

**Comment by ywarnier** [comment]
The checkbox now appears in the registration form. There might still be some issues regarding the registration form (i.e. I need to try a fresh install to see if the notification fields have disappeared), so I'm keeping this opened, but it's not blocking for alpha 3.

**Comment by ywarnier** [comment]
Notification-related fields are now hidden by default in the registration form, so closing.
